tests/plotting/test_widgets.py

```diff
@@ -332,6 +332,82 @@ def test_widget_closed(uniform):
         pl.add_checkbox_button_widget(callback=lambda value: value)
 
 
+def test_widget_radio_button(uniform):
+    p = pv.Plotter()
+    func = lambda: None  # Does nothing
+    p.add_mesh(uniform)
+    b = p.add_radio_button_widget(callback=func, radio_button_group='group')
+    assert p.radio_button_widget_dict['group'][0] == b
+    p.close()
+    assert 'group' not in p.radio_button_widget_dict
+
+
+def test_widget_radio_button_click(uniform):
+    p = pv.Plotter()
+    func = lambda: None  # Does nothing
+    p.add_mesh(uniform)
+    b = p.add_radio_button_widget(
+        callback=func, radio_button_group='group', value=False, position=(10.0, 10.0)
+    )
+    p.show(auto_close=False)
+    # Test switching logic
+    assert b.GetRepresentation().GetState() == 0
+    p.iren._mouse_left_button_click(35, 35)
+    assert b.GetRepresentation().GetState() == 1
+    p.iren._mouse_left_button_click(35, 35)
+    assert b.GetRepresentation().GetState() == 1
+    p.close()
+
+
+def test_widget_radio_button_with_title(uniform):
+    p = pv.Plotter()
+    func = lambda: None  # Does nothing
+    p.add_mesh(uniform)
+    p.add_radio_button_widget(callback=func, radio_button_group='group', title='my_button')
+    assert len(p.radio_button_title_dict['group']) == 1
+    p.close()
+    assert 'group' not in p.radio_button_title_dict
+
+
+def test_widget_radio_button_multiple_on(uniform):
+    p = pv.Plotter()
+    func = lambda: None  # Does nothing
+    p.add_mesh(uniform)
+    b1 = p.add_radio_button_widget(callback=func, radio_button_group='group', value=True)
+    b2 = p.add_radio_button_widget(callback=func, radio_button_group='group', value=True)
+    assert len(p.radio_button_widget_dict['group']) == 2
+    assert b1.GetRepresentation().GetState() == 0
+    assert b2.GetRepresentation().GetState() == 1
+    p.close()
+
+
+def test_widget_radio_button_multiple_switch(uniform):
+    p = pv.Plotter()
+    func = lambda: None  # Does nothing
+    p.add_mesh(uniform)
+    b1 = p.add_radio_button_widget(
+        callback=func, radio_button_group='group', value=True, size=50, position=(10.0, 10.0)
+    )
+    b2 = p.add_radio_button_widget(
+        callback=func, radio_button_group='group', size=50, position=(10.0, 70.0)
+    )
+    p.show(auto_close=False)
+    # Click b1 and switch active radio button
+    p.iren._mouse_left_button_click(35, 95)
```

Comment:
Great test! I have one question. Could you explain how you calculated the button click positions? I’d like to be able to review how this was calculated later on.

Reply:
Good point! First, I just noticed that my comment `# Click b1 ...` is incorrect. The test is clicking 'b2', and checking if active button changes from b1 to b2.

I calculate the button click position as: (button_x + button_size/2, button_y + size/2). (button_x, button_y) is the position of the button passed during button creation, and is the bottom left corner of the button bounding box.

In this test, b2 is defined with `size=50, position=(10.0,70.0)`. So the button click position, or the center of the button, is at `(10 + 50/2, 70 + 50/2) = (35, 95)`.